### PR TITLE
[lld][ELF] Add `--why-extract` for bitcode libcalls

### PR DESCRIPTION
**Pull request by sbc100**
The Wasm linker already records these and its seems useful to do so.

### CODE REVIEW

**Comment by MaskRay** [comment]
Can you add test? You can add `--why-extract=` to a `libcall-*` test

**Comment by MaskRay** [comment]
ping for a test

**Comment by sbc100** [comment]
Sorry, I dropping this completely.. picking it up now.

**Comment by sbc100** [comment]
Test added

File: lld/test/ELF/lto/libcall-archive.ll [hunk 1]
```diff
@@ -4,15 +4,20 @@
...
 ; RUN: llvm-ar rcs %t.a %t2.o %t3.o
-; RUN: ld.lld -o %t %t.o %t.a
+; RUN: ld.lld --why-extract=why.txt -o %t %t.o %t.a
+; RUN: FileCheck %s --input-file=why.txt --check-prefix=CHECK-WHY
 ; RUN: llvm-nm %t | FileCheck %s
 ; RUN: ld.lld -o %t2 %t.o --start-lib %t2.o %t3.o --end-lib
 ; RUN: llvm-nm %t2 | FileCheck %s
 
+; CHECK-WHY: reference extracted symbol
+; CHECK-WHY: <libcall> {{.*}}tmp.a({{.*}}tmp2.o) memcpy
```
Comment (MaskRay) [inline lld/test/ELF/lto/libcall-archive.ll:14]: Add `-NEXT`

**Review by MaskRay — APPROVED** [review]

File: lld/test/ELF/lto/libcall-archive.ll [hunk 2]
```diff
@@ -4,15 +4,20 @@
...
+
 ; CHECK-NOT: T __sync_val_compare_and_swap_8
 ; CHECK: T _start
 ; CHECK: T memcpy
 
+
 target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
```
Comment (MaskRay) [inline lld/test/ELF/lto/libcall-archive.ll:21]: irrelevant change? extra blank line

**Pull request merged**